frontend/src/pages/queryBuilder/QueryGraph.jsx

```diff
@@ -0,0 +1,297 @@
+/* eslint-disable indent, no-use-before-define, func-names, no-return-assign */
+import React, {
+  useEffect, useContext, useRef, useMemo,
+} from 'react';
+import * as d3 from 'd3';
+import graphUtils from '~/utils/d3/graph';
+import nodeUtils from '~/utils/d3/nodes';
+import edgeUtils from '~/utils/d3/edges';
+import highlighter from '~/utils/d3/highlighter';
+
+import BiolinkContext from '~/context/biolink';
+import QueryBuilderContext from '~/context/queryBuilder';
+import getNodeCategoryColorMap from '~/utils/colors';
+import queryGraphUtils from '~/utils/queryGraph';
+
+const nodeRadius = 40;
+
+/**
+ * Main D3 query graph component
+ * @param {int} height - height of the query graph element
+ * @param {int} width - width of the query graph element
+ * @param {obj} clickState - dict of graph click state properties
+ * @param {func} updateClickState - reducer to update graph click state
+ */
+export default function QueryGraph({
+  height, width,
+  clickState, updateClickState,
+}) {
+  const { concepts } = useContext(BiolinkContext);
+  const queryBuilder = useContext(QueryBuilderContext);
+  const nodeCategoryColorMap = useMemo(() => getNodeCategoryColorMap(concepts), [concepts]);
+  const { query_graph } = queryBuilder.state;
+  const queryGraph = useMemo(() => queryGraphUtils.getGraphEditorFormat(query_graph), [queryBuilder.state]);
```

Comment:
I don't think we should have two variables with the same name and different case. Can we come up with something more specific?

Reply:
Updated in 5c44c8c